packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,175 +387,29 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
-                    HTMLAttributes: {
-                        rel: 'noopener noreferrer',
-                        target: null
-                    },
-                    autolink: true,
-                    openOnClick: false,
-                    // linkOnPaste can be enabled when hyperlink support added
-                    // See: https://github.com/ni/nimble/issues/1527
-                    linkOnPaste: false,
-                    validate: href => this.validAbsoluteLinkRegex.test(href)
-                })
-            ]
-        });
-    }
-
-    /**
-     * Extending the default link mark schema defined in the TipTap.
-     *
-     * "excludes": https://prosemirror.net/docs/ref/#model.MarkSpec.excludes
-     * "inclusive": https://prosemirror.net/docs/ref/#model.MarkSpec.inclusive
-     * "parseHTML": https://tiptap.dev/guide/custom-extensions#parse-html
-     * "renderHTML": https://tiptap.dev/guide/custom-extensions/#render-html
-     */
-    private getCustomLinkExtension(): Mark<LinkOptions> {
-        return Link.extend({
-            // Excludes can be removed/enabled when hyperlink support added
-            // See: https://github.com/ni/nimble/issues/1527
-            excludes: '_',
-            // Inclusive can be updated when hyperlink support added
-            // See: https://github.com/ni/nimble/issues/1527
-            inclusive: false,
-            parseHTML() {
-                return [
-                    // To load the `nimble-anchor` from the HTML parsed content by markdown-parser as links in the Tiptap editor, the `parseHTML`
-                    // of Link extension should return nimble `anchorTag`.
-                    // This is because the link mark schema in `markdown-parser.ts` file uses `<nimble-anchor>` as anchor tag and not `<a>`.
-                    {
-                        tag: anchorTag
-                    },
-                    // `<a>` tag is added here to support when pasting a link from external source.
-                    {
-                        tag: 'a'
-                    }
-                ];
-            },
-            // HTMLAttribute cannot be in camelCase as we want to match it with the name in Tiptap
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            renderHTML({ HTMLAttributes }) {
-                // The below 'a' tag should be replaced with 'nimble-anchor' once the below issue is fixed.
-                // https://github.com/ni/nimble/issues/1516
-                return ['a', HTMLAttributes];
-            }
-        });
+    private initializeEditor(): void {
+        this.unbindEditorTransactionEvent();
+        this.unbindEditorUpdateEvent();
+        this.unbindNativeInputEvent();
+        this.tiptapEditor?.destroy();
+        this.tiptapEditor = createTiptapEditor(
+            this.editor,
+            this.configuration?.mentionExtensionConfig ?? []
```

Comment:
Similar to above, we can get rid of the generics complexity by just adding a check:

```
this.configuration instanceof EditorConfiguration ? this.configuration.mentionExtensionConfig : []
```

Reply:
Updated